porch/controllers/packagevariants/pkg/controllers/packagevariant/injection.go

```diff
@@ -0,0 +1,425 @@
+// Copyright 2023 The kpt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packagevariant
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sort"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	kptfilev1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	porchapi "github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	api "github.com/GoogleContainerTools/kpt/porch/controllers/packagevariants/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ConfigInjectionAnnotation  = "kpt.dev/config-injection"
+	InjectedResourceAnnotation = "kpt.dev/injected-resource"
+)
+
+func injectionConditionType(object *fn.KubeObject) string {
+	return strings.Join([]string{"config", "injection", object.GetKind(), object.GetName()}, ".")
```

Comment:
Can't remember if we discussed this elsewhere, but did we decide against including the apiVersion of the object here? I don't have a strong opinion either way, just noticed that it is omitted

Reply:
Yeah, I want these values to be stable across versions. The Kind+Name is sufficient for the package author to avoid ambiguity in the condition types.